index.bs

```diff
@@ -1046,84 +1005,431 @@ BigIntValue = {
+        <dt>-0
+        <dd>Let |serialized| be <code>"-0"</code>
+        <dt>+Infinity
+        <dd>Let |serialized| be <code>"+Infinity"</code>
+        <dt>-Infinity
+        <dd>Let |serialized| be <code>"-Infinity"</code>
+        <dt>Otherwise:
+        <dd>Let |serialized| be |value|
+      </dl>
+
+    1. Let |remote value| be a map matching the <code>NumberValue</code>
+       production in the [=local end definition=], with the <code>value</code>
+       property set to |serialized|.
+
+    <dt>[=Type=](|value|) is Boolean
+    <dd>Let |remote value| be a map matching the <code>BooleanValue</code>
+        production in the [=local end definition=], with the <code>value</code>
+        property set to |value|.
+
+    <dt>[=Type=](|value|) is BigInt
+    <dd>Let |remote value| be a map matching the <code>BigIntValue</code>
+        production in the [=local end definition=], with the <code>value</code>
+        property set to the result of running the [=ToString=] operation on
+        |value|.
+
+  </dl>
+
+1. Return |remote value|
+
+</div>
+
+<div algorithm>
+
+To <dfn>deserialize primitive protocol value</dfn> given a |primitive protocol value|:
+
+1. Let |type| be the value of the <code>type</code> field of |primitive protocol value|.
+
+1. Let |value| be undefined.
+
+1. If |primitive protocol value| has field <code>value</code>:
+   1. Let |value| be the value of the <code>value</code> field of |primitive protocol value|.
+
+1. In the following list of conditions and associated steps, run the first set
+   of steps for which the associated condition is true:
+
+  <dl>
+    <dt>|type| is undefined
+    <dd>Return [=error=] with [=error code=] [=invalid argument=].
+
+    <dt>|type| equals <code>UndefinedValue.type</code>
+    <dd>Return [=success=] with data <code>undefined</code>.
+
+    <dt>|type| equals <code>NullValue.type</code>
+    <dd>Return [=success=] with data <code>null</code>.
+
+    <dt>|type| equals <code>StringValue.type</code>
+    <dd>Return [=success=] with data <code>[=new String=](|value|)</code>.
+
+    <dt>|type| equals <code>NumberValue.type</code>
+    <dd>Return [=success=] with data <code>[=new Number=](|value|)</code>.
+
+    <dt>|type| equals <code>BooleanValue.type</code>
+    <dd>Return [=success=] with data |value|.
+
+    <dt>|type| equals <code>BigIntValue.type</code>
+    <dd>Return [=success=] with data <code>[=new BigInt=](|value|)</code>.
+
+  </dl>
+
+1. Return [=error=] with [=error code=] [=invalid argument=]
+
+</div>
+
+### Local Value ### {#data-types-protocolValue-LocalValue}
+
+Represents both primitive and non-primitive protocol values which can be
+deserialized to ECMAScript without reference to existing objects.
+
+
+```
+LocalValue = {
+  PrimitiveProtocolValue //
+  ArrayLocalValue //
+  DateLocalValue //
+  MapLocalValue //
+  ObjectLocalValue //
+  RegExpLocalValue //
+  SetLocalValue //
+}
+
+ListLocalValue = [*LocalValue];
+
+ArrayLocalValue = {
+  type: "array",
```

Comment:
So for serialization we have `typedarray` and `arraybuffer`. How does that map to deserialization? I don't see that we could deserialize such RemoteValue instances.

Also why do we need two different (remote/local) types for the same RemoteValue definition? That's not clear to me and I haven't seen a discussion about that yet. In general `serialize()` produces a `RemoteValue` and deserialize takes exactly such an object as input.